doc/releases/release-notes-2.6.rst

```diff
@@ -184,6 +186,12 @@ Drivers and Sensors
 
 * Flash
 
+  * flash_write_protection_set() was deprecated API and become no-operation.
```

Comment:
While this isn't wrong, it confused me because it only explains part of what has been and what needs to be done. 
 People reading this may not know what they need to do to update out-of-tree drivers and their code.

I drafted this which is closer, but probably not quite right.  It should be reconciled with what's in the other comment in this review.

> `flash_write_protection_set()` has been deprecated and will be removed in Zephyr 2.8.  Responsibility for write/erase protection management has been moved to the driver-specific implementation of the `flash_write()` and `flash_erase()` API calls.  All in-tree flash drivers have been updated, and the protect implementation removed from their API tables. During the deprecation period user code invoking `flash_write_protection_set()` will have no effect, but the `flash_write()` and `flash_erase()` driver shims will wrap their calls with calls to the protect implementation if it is present in the API table.  Out-of-tree drivers must be updated before the wrapping in the shims is removed when the deprecation period ends.

Reply:
Applied